packages/core/src/components/icon/icon.tsx

```diff
@@ -7,60 +7,73 @@
 import * as classNames from "classnames";
 import * as React from "react";
 
-import { IconName } from "@blueprintjs/icons";
+import { IconName, IconSvgPaths16, IconSvgPaths20, LegacyIconName } from "@blueprintjs/icons";
 import { Classes, IIntentProps, IProps } from "../../common";
 
 export { IconName };
 
 export interface IIconProps extends IIntentProps, IProps {
+    /**
+     * Color of icon. Equivalent to setting CSS `fill` property.
+     */
+    color?: string;
+
     /**
      * Name of the icon (with or without `"pt-icon-"` prefix).
-     * If `undefined`, this component will render nothing.
+     * If omitted or `undefined`, this component will render nothing.
      */
-    iconName: IconName | undefined;
+    iconName?: LegacyIconName;
 
     /**
-     * Size of the icon.
-     * Blueprint provides each icon in two sizes: 16px and 20px. The keyword `"inherit"` will
-     * render a 20px icon but inherit `font-size` from its parent.
-     * Constants are exposed for each of these values on the component itself:
-     * `Icon.SIZE_(STANDARD|LARGE|INHERIT)`,
-     * @default 16
+     * Size of the icon, in pixels.
+     * Blueprint contains 16px and 20px SVG icon images,
+     * and chooses the appropriate resolution based on this prop.
+     * @default Icon.SIZE_STANDARD = 16
      */
-    iconSize?: 16 | 20 | "inherit";
+    iconSize?: number;
+
+    /** CSS style properties. */
+    style?: React.CSSProperties;
 }
 
-export class Icon extends React.PureComponent<IIconProps & React.HTMLAttributes<HTMLSpanElement>, never> {
+export class Icon extends React.PureComponent<IIconProps & React.SVGAttributes<SVGElement>> {
     public static displayName = "Blueprint2.Icon";
 
-    public static readonly SIZE_STANDARD = 16 as 16;
-    public static readonly SIZE_LARGE = 20 as 20;
-    public static readonly SIZE_INHERIT = "inherit" as "inherit";
+    public static readonly SIZE_STANDARD = 16;
+    public static readonly SIZE_LARGE = 20;
 
     public render() {
         if (this.props.iconName == null) {
             return null;
         }
-        const { className, iconName, intent, iconSize = Icon.SIZE_STANDARD, ...restProps } = this.props;
+        const { className, iconName: iconNameProp, iconSize = Icon.SIZE_STANDARD, intent, ...svgProps } = this.props;
+        const iconName = iconNameProp.replace("pt-icon-", "") as IconName;
```

Comment:
nit: leave the prop as `iconName` (also do this destructuring on the first line of the function, so you can use it on L46) and make this variable named `normalizedIconName`